iaso/api/payments/views.py

```diff
@@ -476,19 +475,13 @@ def calculate_new_potential_payments(self):
                 potential_payment__task__isnull=True,
             )
             if change_requests.exists():
-                try:
-                    potential_payment, created = PotentialPayment.objects.get_or_create(
-                        user_id=user["created_by"],
-                    )
-                    for change_request in change_requests:
-                        change_request.potential_payment = potential_payment
-                        change_request.save()
-                    potential_payment.save()
-                # If there was a race condition, we return the existing PotentialPayment
-                except IntegrityError:
-                    return PotentialPayment.objects.get(
-                        user_id=user["created_by"],
-                    )
+                potential_payment, created = PotentialPayment.objects.get_or_create(
+                    user_id=user["created_by"],
+                )
+                for change_request in change_requests:
+                    change_request.potential_payment = potential_payment
+                    change_request.save()
+                potential_payment.save()
```

Comment:
@tdethier @quang-le I've removed the exception catching here because [`get_or_create`](https://docs.djangoproject.com/en/5.1/ref/models/querysets/#get-or-create) is atomic when the database enforces uniqueness of the keyword arguments (`user` here).

And now, the uniqueness constraint is set by the [`OneToOneField`](https://github.com/django/django/blob/857b1048d53ebf5fc5581c110e85c212b81ca83a/django/db/models/fields/related.py#L1218-L1221).

`IntegrityError` is then [catched in `get_or_create`](https://github.com/django/django/blob/857b1048d53ebf5fc5581c110e85c212b81ca83a/django/db/models/query.py#L945C1-L946C1).

Are you OK with that?

Reply:
Yup I'm sure about that.